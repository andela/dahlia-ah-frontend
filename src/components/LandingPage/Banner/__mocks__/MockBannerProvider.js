import React from 'react';
import PropTypes from 'prop-types';
import { BannerContext } from '../../../../context/BannerContext';

export const mockBannerContext = {
  banner: [
    {
      author: 'James Bond',
      coverImgUrl: 'https://res.cloudinary.com/allebd/image/upload/v1567555883/dahlia/back9unsplash.jpg',
      createdAt: '2019-09-11T18:20:55.915Z',
      description: 'Our journey to higher heights',
      genre: 'thriller',
      id: 'b45bdf8c-a9d5-436c-b2f6-f5c0fa56596a',
      isPublished: true,
      slug: 'higher-and-higher',
      thumbImgUrl: 'https://res.cloudinary.com/allebd/image/upload/v1567524085/dahlia/abstractbookcover.png',
      title: 'Higher and Higher',
      updatedAt: '2019-09-11T18:20:55.915Z',
    },
  ],
};

const MockBannerProvider = ({ children }) => {
  const { Provider } = BannerContext;
  return (
    <Provider value={{ ...mockBannerContext }}>
      {children}
    </Provider>
  );
};

MockBannerProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MockBannerProvider;
