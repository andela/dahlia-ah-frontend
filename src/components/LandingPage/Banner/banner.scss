/*
* banner.scss 
* 
* Dahlia Frontend - Andela Simulations Challenge ©️ 2019
* Andela Simulations Challenge ©️ 2019
* https://github.com/andela/dahlia-ah-frontend
*
* SCSS Designed by TeamDahlia.
*/

$grey: #e5e5e5;
$white: #ffffff;
$black: #000008;

/* Banner */
#banner {
  background: url(https://res.cloudinary.com/allebd/image/upload/v1567031693/dahlia/banner.png);
  background-repeat: no-repeat;
  background-position: center top;
  background-size: 100% auto;
  height: 800px;
  color: $white;

  .bannerGradient {
    width: 100%;
    height: 100%;
    overflow: hidden;
    background: rgba(0,0,0,0.3);
    box-shadow: 0px 0px 40px 40px rgba(0,0,0,0.3);
    align-items: center;
    padding: 4em;
    padding-top: 16em;

    h2 {
      font-size: 3.5em;
      line-height: 1.2;
      text-transform: uppercase;
      font-weight: bold;
      margin: 0;
    }

    .bannerText {
      width: 500px;
    }
  }
}

@media(max-width: 1200px){
  #banner {
    background-size: auto;
    height: 100%;
  }
}

@media(max-width: 900px){
  #banner {
    .bannerGradient {
      width: 100%;
      padding-top: 30%;
      
      .bannerText {
        width: 80%;
      }
    }
  }
}
