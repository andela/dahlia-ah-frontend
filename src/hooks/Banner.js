import { useState } from 'react';

const bannerState = () => {
  const [banner, setBanner] = useState('');
  return { banner, setBanner };
};

export default bannerState;
