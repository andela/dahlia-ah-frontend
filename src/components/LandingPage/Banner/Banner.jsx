import React from 'react';
import PropType from 'prop-types';
import './banner.scss';

const Banner = ({ openModal }) => (
  <div>
    <section id="banner">
      <div className="bannerGradient">
        <h2>THE MAN IN THE FOREST OF DREAMS</h2>
        <div className="bannerText">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,
            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco
            laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure
            dolor in reprehenderit in voluptate velit esse cillum dolore
            eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
            non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
        <button type="button" onClick={() => { openModal('signin'); }} className="btnLarge">View Snippet</button>
      </div>
    </section>
  </div>
);

Banner.propTypes = {
  openModal: PropType.func.isRequired,
};


export default Banner;
