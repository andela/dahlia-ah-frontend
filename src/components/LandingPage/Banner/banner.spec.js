import React from 'react';
import { configure, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import Banner from './Banner';
import MockBannerProvider from './__mocks__/MockBannerProvider';

configure({ adapter: new Adapter() });

describe('Banner component', () => {
  let wrapper;
  const mockFunction = jest.fn();

  beforeEach(() => {
    wrapper = mount(
      <MockBannerProvider>
        <Banner openModal={() => {}} />
      </MockBannerProvider>,
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('modal opens on clicking section image', () => {
    wrapper.find('section').simulate('click');
    expect(mockFunction.mock.calls.length).toEqual(0);
  });

  it('modal opens on clicking banner title', () => {
    wrapper.find('h2').simulate('click');
    expect(mockFunction.mock.calls.length).toEqual(0);
  });

  it('modal opens on clicking see more button', () => {
    wrapper.find('button').simulate('click');
    expect(mockFunction.mock.calls.length).toEqual(0);
  });
});
