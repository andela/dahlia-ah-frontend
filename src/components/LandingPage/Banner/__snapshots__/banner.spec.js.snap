// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `ReactWrapper {}`;
